pkg/experiment/sensitivity/workload_factory.go

```diff
@@ -0,0 +1,276 @@
+	)
+
+	theatAggressorsAsService = conf.NewBoolFlag(
+		"debug_treat_be_as_service", "Debug only: Best Effort workloads are wrapped in Service flags so that the experiment can track their lifectcle. Default `true` should not be changed without explicit reason.",
+		true)
+
+	// L1dProcessNumber represents number of L1 data cache aggressor processes to be run
+	L1dProcessNumber = conf.NewIntFlag(
+		"experiment_be_l1d_processes_number",
+		"Number of L1 data cache best effort processes to be run",
+		l1dDefaultProcessNumber,
+	)
+
+	// L1iProcessNumber represents number of L1 instruction cache aggressor processes to be run
+	L1iProcessNumber = conf.NewIntFlag(
+		"experiment_be_l1i_processes_number",
+		"Number of L1 instruction cache best effort processes to be run",
+		l1iDefaultProcessNumber,
+	)
+
+	// L3ProcessNumber represents number of L3 data cache aggressor processes to be run
+	L3ProcessNumber = conf.NewIntFlag(
+		"experiment_be_l3_processes_number",
+		"Number of L3 data cache best effort processes to be run",
+		l3DefaultProcessNumber,
+	)
+
+	// MembwProcessNumber represents number of membw aggressor processes to be run
+	MembwProcessNumber = conf.NewIntFlag(
+		"experiment_be_membw_processes_number",
+		"Number of membw best effort processes to be run",
+		membwDefaultProcessNumber,
+	)
+)
+
+// WorkloadFactory is creator for High Priority and Best Effort workloads with
+// default or custom isolation.
+type WorkloadFactory struct {
+	executorFactory ExecutorFactory
+
+	hpIsolation isolation.Decorator
+	l1Isolation isolation.Decorator
+	l3Isolation isolation.Decorator
+}
+
+// NewDefaultWorkloadFactory returns factory that would create Workloads on Kubernetes executor
+// or Local executor, depending on flags.
+func NewDefaultWorkloadFactory() WorkloadFactory {
+	return NewWorkloadFactory(NewExecutorFactory())
+}
+
+// NewWorkloadFactory creates new instance of WorkloadFactory.
+func NewWorkloadFactory(executorFactory ExecutorFactory) WorkloadFactory {
+	hpIsolation, l1Isolation, l3Isolation := GetWorkloadsIsolations()
+
+	return NewWorkloadFactoryWithIsolation(executorFactory, hpIsolation, l1Isolation, l3Isolation)
+}
+
+// NewWorkloadFactoryWithIsolation creates new workload factory with custom isolation.
+func NewWorkloadFactoryWithIsolation(
+	factory ExecutorFactory,
+	hpIsolation isolation.Decorator,
+	l1Isolation isolation.Decorator,
+	l3Isolation isolation.Decorator) WorkloadFactory {
+	return WorkloadFactory{
+		executorFactory: factory,
+		hpIsolation:     hpIsolation,
+		l1Isolation:     l1Isolation,
+		l3Isolation:     l3Isolation,
+	}
+}
+
+// BuildDefaultHighPriorityLauncher builds High Priority workload launcher with predefined isolation.
+func (factory *WorkloadFactory) BuildDefaultHighPriorityLauncher(
+	workloadName string, tags snap.Tags) (launcher executor.Launcher, err error) {
+	return factory.BuildHighPriorityLauncherWithIsolation(workloadName, factory.hpIsolation, tags)
```

Comment:
@skonefal - the only thing that bothers me in this case is increased code complexity; I would prefer to keep it is simple as possible. 
In case of constructors there are only two functions and one of them includes some logic (generated default configuration values). Here we have an additional intermediary. I would appreciate if you dropped it.